File: LeetCode/70. Climbing Stairs/elixir/solution.exs
defmodule Solution do
  @spec climb_stairs(n :: integer) :: integer
  def climb_stairs(n), do: climb(n, 1, 0)
  defp climb(0, l, _), do: l
  defp climb(n, l, r), do: climb(n - 1, l + r, l)
end

File: LeetCode/70. Climbing Stairs/java/Solution.java
class Solution {
    public int climbStairs(int n) {
        if (n <= 3) {
            return n;
        }
        int[] dp = new int[n];
        dp[2] = 2;
        dp[3] = 3;
        for (int i = 4; i < n; i++) {
            dp[i] = dp[i - 1] + dp[i - 2];
        }
        return dp[n - 2] + dp[n - 1];
    }
}
